File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        // get float number
        dollars = get_float("how much change is owed: ");
    }
    // check if dollars is non negative
    while (dollars < 0);
    // round dollars as integer
    int cents = round(dollars * 100);
    // check how many coins require
    int quaters = cents / 25;
    int dimes = (cents % 25) / 10;
    int nickels = ((cents % 25) % 10) / 5;
    int pennies = (((cents % 25) % 10) % 5);

    printf("%d\n", quaters + dimes + nickels + pennies);
    
}

File: cash.py
coins = [0.25, 0.1, 0.05, 0.01]
count = 0

while True:
    change = input("Change owed: ")
    if change.replace(".", "").isdigit() == True:
        if float(change) >= 0:
            break

change = float(change)
for i in range(len(coins)):
    while change > coins[i] or change == coins[i]:
            multipy = int(change / coins[i])
            change = round(change % coins[i], 2)
            count += multipy
    if change == 0:
        break

print(f"{count}")




